src/components/LHNOptionsList/OptionRowLHN.js

```diff
@@ -0,0 +1,220 @@
+import _ from 'underscore';
+import lodashGet from 'lodash/get';
+import React from 'react';
+import PropTypes from 'prop-types';
+import {
+    TouchableOpacity,
+    View,
+    StyleSheet,
+} from 'react-native';
+import styles from '../../styles/styles';
+import * as StyleUtils from '../../styles/StyleUtils';
+import Icon from '../Icon';
+import * as Expensicons from '../Icon/Expensicons';
+import MultipleAvatars from '../MultipleAvatars';
+import Hoverable from '../Hoverable';
+import DisplayNames from '../DisplayNames';
+import IOUBadge from '../IOUBadge';
+import colors from '../../styles/colors';
+import withLocalize, {withLocalizePropTypes} from '../withLocalize';
+import Text from '../Text';
+import SubscriptAvatar from '../SubscriptAvatar';
+import CONST from '../../CONST';
+import * as ReportUtils from '../../libs/ReportUtils';
+import variables from '../../styles/variables';
+import themeColors from '../../styles/themes/default';
+import OptionsListUtilsLHN from '../../libs/OptionsListUtilsLHN';
+
+const propTypes = {
+    /** Style for hovered state */
+    // eslint-disable-next-line react/forbid-prop-types
+    hoverStyle: PropTypes.object,
+
+    /** The ID of the report that the option is for */
+    reportID: PropTypes.string.isRequired,
+
+    /** Whether this option is currently in focus so we can modify its style */
+    optionIsFocused: PropTypes.bool,
+
+    /** A function that is called when an option is selected. Selected option is passed as a param */
+    onSelectRow: PropTypes.func,
+
+    /** Toggle between compact and default view */
+    viewMode: PropTypes.oneOf(_.values(CONST.OPTION_MODE)),
+
+    style: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.object), PropTypes.object]),
+
+    ...withLocalizePropTypes,
+};
+
+const defaultProps = {
+    hoverStyle: styles.sidebarLinkHover,
+    viewMode: 'default',
+    onSelectRow: () => {},
+    optionIsFocused: false,
+    style: null,
+};
+
+const OptionRowLHN = (props) => {
+    const optionItem = OptionsListUtilsLHN.getOptionData(props.reportID);
```

Comment:
It's doing some thing for every row. For instance getPersonalDetailsForLogins. Maybe we could memoize that?

Reply:
![Screenshot 2022-09-05 at 15 21 37](https://user-images.githubusercontent.com/12784455/188459535-68c38e93-5e83-4667-aa54-c330c14900cd.png)

Reply:
I'm torn on using memoization for this. There is so much data which depends on being present for `getOptionData()` and it all changes at different rates, so I am worried about data becoming stale if it's memoized. So, I think before doing that, I'd rather make this code as efficient as we can. For example, I think this can still be optimized:

```
    const personalDetailMap = OptionsListUtils.getPersonalDetailsForLogins(report.participants, personalDetails);
    const personalDetailList = _.values(personalDetailMap);
    const personalDetail = personalDetailList[0];
```

I haven't started on this yet, but we are only using the first element of `personalDetailList` which means we don't need to pass all of `report.participants`. I also think we can avoid `_.values()` and probably some of the code in `getPersonalDetailsForLogins`